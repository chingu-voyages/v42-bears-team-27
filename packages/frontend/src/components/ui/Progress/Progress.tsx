import type { ForwardRefExoticComponent, RefAttributes } from 'react';
import { forwardRef } from 'react';
import type { ThemeUIStyleObject } from 'theme-ui';
import * as ProgressPrimitive from '@radix-ui/react-progress';

export interface ProgressProps extends ProgressPrimitive.ProgressProps {
  sx?: ThemeUIStyleObject;
}

export const Progress: ForwardRefExoticComponent<
ProgressProps & RefAttributes<HTMLDivElement>
> = forwardRef(({ value, sx, ...rest }) => (
  <ProgressPrimitive.Root
    value={value}
    sx={{
      position: 'relative',
      overflow: 'hidden',
      bg: 'background',
      borderRadius: 6,
      width: 300,
      height: 30,
      transform: 'translateZ(0)',
      ...sx,
    }}
    {...rest}
  >
    <ProgressPrimitive.Indicator
      sx={{
        width: '100%',
        height: '100%',
        bg: 'primary',
        border: 'none',
        transform: `translateX(-${100 - Number(value)}%)`,
        transition: 'transform 660ms cubic-bezier(0.65, 0, 0.35, 1)',
      }}
    />
  </ProgressPrimitive.Root>
));

if (process.env.NODE_ENV !== 'production') {
  Progress.displayName = 'Progress';
}
