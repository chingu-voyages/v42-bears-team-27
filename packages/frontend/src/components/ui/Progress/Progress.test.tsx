import { render } from 'test-utils';

import { Progress } from './Progress';

describe('<Progress />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Progress />);
      expect(container).toMatchSnapshot();
    });
  });
});
