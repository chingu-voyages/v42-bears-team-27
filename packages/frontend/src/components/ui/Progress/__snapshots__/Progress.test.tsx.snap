// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> Snapshots should match snapshot 1`] = `
.emotion-0 {
  position: relative;
  overflow: hidden;
  background-color: var(--theme-ui-colors-background);
  border-radius: 6px;
  width: 300px;
  height: 30px;
  -webkit-transform: translateZ(0);
  -moz-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
}

.emotion-1 {
  width: 100%;
  height: 100%;
  background-color: var(--theme-ui-colors-primary);
  border: none;
  -webkit-transform: translateX(-NaN%);
  -moz-transform: translateX(-NaN%);
  -ms-transform: translateX(-NaN%);
  transform: translateX(-NaN%);
  -webkit-transition: -webkit-transform 660ms cubic-bezier(0.65, 0, 0.35, 1);
  transition: transform 660ms cubic-bezier(0.65, 0, 0.35, 1);
}

<div>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    class="emotion-0"
    data-max="100"
    data-state="indeterminate"
    role="progressbar"
  >
    <div
      class="emotion-1"
      data-max="100"
      data-state="indeterminate"
    />
  </div>
</div>
`;
